app/code/Magento/Cms/Controller/Adminhtml/Block/Save.php

```diff
@@ -55,27 +73,33 @@ public function execute()
             }
 
             /** @var \Magento\Cms\Model\Block $model */
-            $model = $this->_objectManager->create(\Magento\Cms\Model\Block::class)->load($id);
-            if (!$model->getId() && $id) {
-                $this->messageManager->addError(__('This block no longer exists.'));
-                return $resultRedirect->setPath('*/*/');
+            $model = $this->blockFactory->create();
+
+            $id = $this->getRequest()->getParam('block_id');
+            if ($id) {
+                $model->load($id);
```

Comment:
Could we not also load the block from the repository?

Reply:
Something like `$this->blockRepository->getById($id)`

Reply:
Sure, give me 2 sec.